core/src/main/java/org/apache/iceberg/io/V2TaskWriter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import java.io.Closeable;
+import java.io.IOException;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.StructLike;
+
+public interface V2TaskWriter<T> extends Closeable {
+
+  void insert(T row, PartitionSpec spec, StructLike partition) throws IOException;
+
+  void abort() throws IOException;
```

Comment:
What is the expectation for calling abort? Should the writer already be closed?

Reply:
This essentially mimics the current interface we have. I am happy to reconsider having `abort` in our writers. I guess the idea is to reuse the cleanup logic in all query engine integrations. We could have some utility methods instead.

Reply:
I removed `abort` and it looks like we don't really need `V2TaskWriter` anymore. So I removed that interface completely and just kept `DeltaWriter`.